static/app/components/events/interfaces/performance/spanEvidence.tsx

```diff
@@ -50,16 +59,51 @@ export function SpanEvidenceSection({event, organization}: Props) {
       )}
     >
       <SpanEvidenceKeyValueList event={event} />
-
-      <TraceViewWrapper>
-        <TraceView
-          organization={organization}
-          waterfallModel={
-            new WaterfallModel(event as EventTransaction, affectedSpanIds, focusedSpanIds)
-          }
-          isEmbedded
-        />
-      </TraceViewWrapper>
+      {hasProfilingPreviewsFeature ? (
+        <ProfilesProvider
+          orgSlug={organization.slug}
+          projectSlug={projectSlug}
+          profileId={profileId || ''}
+        >
+          <ProfileContext.Consumer>
+            {profiles => (
+              <ProfileGroupProvider
+                type="flamechart"
+                input={profiles?.type === 'resolved' ? profiles.data : null}
+                traceID={profileId || ''}
+              >
+                <TraceViewWrapper>
+                  <TraceView
+                    organization={organization}
+                    waterfallModel={
+                      new WaterfallModel(
```

Comment:
I guess this is kinda inefficient as it'll recreate this on each render, but it was there before so assuming it's small enough to not care 🤷🏼

Reply:
Wasn't thinking about this but this is definitely not a trivial operation as it constructs the span waterfall from the transaction each time. Gonna memo this.